src/python/WMCore/WMSpec/ConfigSectionTree.py

```diff
@@ -227,7 +227,7 @@ def format(value):
     format a value as python
     keep parameters simple, trust python...
     """
-    if type(value) == str:
+    if isinstance(value, str):
```

Comment:
Won't we get ourselves into big trouble if value is unicode? I mean, don't we need to have future imports here?

Reply:
What makes this PR contain trivial changes is that `type(value) == str` and `isinstance(value, str)` are supposed to give the same result, whatever is the type of `value`. 

We already have https://github.com/dmwm/WMCore/blob/979f610f0e62b61bcdc4cea68c2b2726a6d26d5c/src/python/WMCore/WMSpec/ConfigSectionTree.py#L13
so that in this line `str` is already a sequence of unicode codepoints. 

But there is no such thing as a "trivial" change and quirks are hidden everywhere. If value is a sequence of unicode codepoints, then in py3 it is of type `str`, so that both `type(value) == str` and `isinstance(value, str)` will likely return the same result in very situation. 
Meanwhile, in py2 it can get a bit confusing. This few lines summarize what could happen

```plaintext
Python 2.7.13 (default, Sep 14 2017, 23:43:58) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a = u'ciao'
>>> type(a)
<type 'unicode'>
>>> from builtins import str // str becomes <class 'future.types.newstr.newstr'>
>>> type(a) == str
False
>>> isinstance(a, str)
True
```

It is possible that `type(a) == str` incorrectly returns `False` when `a` is a string of type unicode codepoints sequence.

Then, you can wonder "what if value is a standard py2 string of type `str`?" Well, I just realized we have been a bit imprecise in #10103. Before adding `from builtins import str`, L230 would have matched standard py2 `str` and added `'` around them (unicode strings were not changed). After adding it, only the new type of py2 `str` introduced by future would get this (unicode strings would be changed, bytes strings would not). We changed the behavior of our code without noticing. Impact of this change: the only place where this change could have affected us is in https://github.com/dmwm/WMCore/blob/979f610f0e62b61bcdc4cea68c2b2726a6d26d5c/src/python/WMCore/WMSpec/WMTask.py#L631 , but `WMTask_t` has no unit tests for it. If we want to be sure this did not cause problems we would have to continue following where this function is used downstream (such as checking `./src/python/WMCore/JobSplitting/Generators/GeneratorFactory.py`).
We have to options here: remove `from builtins import str` and continue formatting `str` in both py2 and py3, despite these are two different things, or properly format both unicode and bytes strings in both py2 and py3. 

On a sidenote: Doesn't the line https://github.com/dmwm/WMCore/blob/979f610f0e62b61bcdc4cea68c2b2726a6d26d5c/src/python/WMCore/WMSpec/ConfigSectionTree.py#L248 feel wrong, does it?

Reply:
> We have to options here: remove from builtins import str and continue formatting str in both py2 and py3, despite these are two different things, or properly format both unicode and bytes strings in both py2 and py3.

I'm not sure what's the best way to proceed here, and I would like to avoid pulling actions from other slices as much as we can (likely properly dealing with unicode and byte strings here, as you suggested at the very end).
I guess a third option would be not to change anything else and see how tests go(?) 

> On a sidenote: Doesn't the line 

It does! I think we should change it to `return value` to keep it consistent with lists, for instance. However, I don't think we have ever had such data types there, otherwise things would have likely had broken already.

Reply:
>  I would like to avoid pulling actions from other slices as much as we can

I totally agree. What I propose is that I open a new issue about "possible bug introduced in #10103 " ans I put it in python3 project in the column "WMCore py2->py2py3: bugs and follow up". I can spend a couple of hours to see what the impact of that bug could be and then we try to make the bug happen when we cut the test deploy.

> However, I don't think we have ever had such data types there, otherwise things would have likely had broken already.

yeah, it's been sitting there since 11 years! I like your suggestion of changing it eventually!

Reply:
Having an extra GH issue with a possible problem is a good idea, indeed.

Reply:
(I made a new commit with `return dict` -> `return value`)